fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -515,8 +515,8 @@ public FedoraResource findOrCreateAcl() {
                 LOGGER.debug("Created Webac ACL {}", aclNode.getPath());
 
                 // add mixin type fedora:Resource
-                if (node.canAddMixin(FEDORA_RESOURCE)) {
-                    node.addMixin(FEDORA_RESOURCE);
+                if (aclNode.canAddMixin(FEDORA_RESOURCE)) {
```

Comment:
This update may be the cause of the issue noted in the comment:
https://jira.duraspace.org/browse/FCREPO-2810?focusedCommentId=60115&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-60115